var/spack/repos/builtin/packages/sensei/package.py

```diff
@@ -0,0 +1,104 @@
+    version('1.0.0', sha256='bdcb03c56b51f2795ec5a7e85a5abb01d473d192fac50f2a8bf2608cc3564ff8')
+
+    variant('build_type', default='RelWithDebInfo', description='CMake build type', values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
+    variant('sencore', default=True, description='Enables the SENSEI core library')
+    variant('catalyst', default=True, description='Build with ParaView-Catalyst support')
+    variant('libsim', default=False, description='Build with VisIt-Libsim support')
+    variant('vtkio', default=True, description='Enable adaptors to write to VTK XML format')
+    variant('adios', default=False, description='Enable ADIOS adaptors and endpoints')
+    variant('python', default=False, description='Enable Python bindings')
+    variant('miniapps', default=True, description='Enable the parallel 3D and oscillators miniapps')
+
+    depends_on("paraview@5.5:5.6", when="+catalyst")
+    depends_on("visit", when="+libsim")
+    depends_on("vtk", when="+libsim")
+    depends_on("vtk", when="+adios ~libsim ~catalyst")
+    depends_on("vtk", when="+sencore ~catalyst")
+    depends_on("vtk", when="+python ~libsim ~catalyst")
+    depends_on("adios", when="+adios")
+    depends_on("python", when="+python")
+    depends_on("py-numpy", when="+python")
+    depends_on("py-mpi4py", when="+python")
+    depends_on("swig", when="+python")
+
+    # Can have either LibSim or Catalyst, but not both
+    conflicts('+libsim', when='+catalyst')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            '-DCMAKE_CXX_STANDARD=11',
+            '-DCMAKE_C_STANDARD=11',
+            '-DCMAKE_CXX_FLAGS=-fPIC -Wall -Wextra -O3 -mtune=generic',
+            '-DCMAKE_C_FLAGS=-fPIC -Wall -Wextra -O3 -mtune=generic'
+        ]
+
+        sensei_switch = 'ON' if '+sencore' in spec else 'OFF'
+        args.append('-DENABLE_SENSEI:BOOL={0}'.format(sensei_switch))
+
+        vtk_dir_needed = True
+
+        if '+catalyst' in spec:
+            args.append('-DENABLE_CATALYST:BOOL=ON')
+            args.append('-DENABLE_CATALYST_PYTHON:BOOL=ON')
+            pview_prefix = spec['paraview'].prefix
+            args.append('-DParaView_DIR:PATH={0}'.format(pview_prefix))
+            vtk_dir_needed = False
+        else:
+            args.append('-DENABLE_CATALYST:BOOL=OFF')
+
+        if '+libsim' in spec:
+            args.append('-DENABLE_LIBSIM:BOOL=ON')
+            args.append('-DVISIT_DIR:PATH={0}'.format(spec['visit'].prefix))
+            args.append('-DVTK_DIR:PATH={0}'.format(spec['vtk'].prefix))
+            vtk_dir_needed = False
+        else:
+            args.append('-DENABLE_LIBSIM:BOOL=OFF')
+
+        vtkio_switch = 'ON' if '+vtkio' in spec else 'OFF'
+        args.append('-DENABLE_VTK_IO:BOOL={0}'.format(vtkio_switch))
+        python_switch = 'ON' if '+python' in spec else 'OFF'
+        args.append('-DENABLE_PYTHON:BOOL={0}'.format(python_switch))
```

Comment:
Is there any flag to tell CMake which python interpreter to use? Just want to make sure this works for both Python 2 and 3 regardless of the interpreter name.

Reply:
When '+python' is enabled Sensei will use a CMake script that searches for Python 2. It looks like handling Python 3 is a TODO.

Reply:
In that case, I would make it explicit by saying `depends_on('python@:2', when='+python')`. It's a shame that Python 3 isn't supported when Python 2 will be EOL by the end of the year.

Reply:
You may want to also add `'-DPYTHON_EXECUTABLE:FILEPATH=' + spec['python'].command.path` if there ends up being problems with the right python getting found.